dev/io.openliberty.jakartaee9.internal_fat/fat/src/io/openliberty/jakartaee9/internal/tests/EE9FeatureCompatibilityTest.java

```diff
@@ -0,0 +1,311 @@
+    public static LibertyServer server;
+
+    @BeforeClass
+    public static void setUp() throws Exception {
+        File featureDir = new File(server.getInstallRoot() + "/lib/features/");
+        // If there was a problem building projects before this test runs, "lib/features" won't exist
+        if (featureDir != null && featureDir.exists()) {
+            for (File feature : featureDir.listFiles()) {
+                parseShortName(feature);
+            }
+        }
+
+        nonEE9JavaEEFeatures.addAll(EE7FeatureReplacementAction.EE7_FEATURE_SET);
+        nonEE9JavaEEFeatures.addAll(EE8FeatureReplacementAction.EE8_FEATURE_SET);
+
+        // appSecurity-1.0 is superceded by appSecurity-2.0 so it isn't in one of the replacement
+        // feature lists.  jaxb and jaxws 2.3 are EE related, but are noship features currently.
+        // jsp-2.2 is a EE6 feature that is included with open liberty.
+        nonEE9JavaEEFeatures.add("appSecurity-1.0");
+        nonEE9JavaEEFeatures.add("jaxb-2.3");
+        nonEE9JavaEEFeatures.add("jaxws-2.3");
+        nonEE9JavaEEFeatures.add("jsp-2.2");
+
+        // Remove test features that are in the FeatureReplacementActions
+        nonEE9JavaEEFeatures.remove("componenttest-1.0");
+        nonEE9JavaEEFeatures.remove("componenttest-2.0");
+        nonEE9JavaEEFeatures.remove("txtest-1.0");
+        nonEE9JavaEEFeatures.remove("txtest-2.0");
+        nonEE9JavaEEFeatures.remove("ejbTest-1.0");
+        nonEE9JavaEEFeatures.remove("ejbTest-2.0");
+
+        nonEE9MicroProfileFeatures.addAll(MicroProfileActions.MP10.getFeatures());
+        nonEE9MicroProfileFeatures.addAll(MicroProfileActions.MP12.getFeatures());
+        nonEE9MicroProfileFeatures.addAll(MicroProfileActions.MP13.getFeatures());
+        nonEE9MicroProfileFeatures.addAll(MicroProfileActions.MP14.getFeatures());
+        nonEE9MicroProfileFeatures.addAll(MicroProfileActions.MP20.getFeatures());
+        nonEE9MicroProfileFeatures.addAll(MicroProfileActions.MP21.getFeatures());
+        nonEE9MicroProfileFeatures.addAll(MicroProfileActions.MP22.getFeatures());
+        nonEE9MicroProfileFeatures.addAll(MicroProfileActions.MP30.getFeatures());
+        nonEE9MicroProfileFeatures.addAll(MicroProfileActions.MP32.getFeatures());
+        nonEE9MicroProfileFeatures.addAll(MicroProfileActions.MP33.getFeatures());
+        nonEE9MicroProfileFeatures.addAll(MicroProfileActions.MP40.getFeatures());
+        nonEE9MicroProfileFeatures.addAll(MicroProfileActions.MP41.getFeatures());
+
+        // Add the MP convenience features as well
+        String[] mpVersions = { "1.0", "1.2", "1.3", "1.4", "2.0", "2.1", "2.2", "3.0",
+                                "3.2", "3.3", "4.0", "4.1" };
+        for (String mpVersion : mpVersions) {
+            nonEE9MicroProfileFeatures.add("microProfile-" + mpVersion);
+        }
+
+        // MP standalone features
+        nonEE9MicroProfileFeatures.add("mpContextPropagation-1.0");
```

Comment:
Is there an authoritative list of these standalone MP features?  Perhaps better maintained in MicroProfileActions?